test-suites/test-main/access-checks.specs/access-currency.cov.spec.ts

```diff
@@ -23,24 +24,46 @@ makeSuite('access: Collateral Currency', (testEnv: TestEnv) => {
     await setInsurer(state, deployer, deployer.address);
   });
 
+  it('ROLE: Unregister without a role', async () => {
+    await expect(state.cc.unregister(state.fund.address)).to.be.reverted;
+    await expect(state.cc.unregister(state.insurer.address)).to.be.reverted;
+    await expect(state.cc.unregister(zeroAddress())).to.be.reverted;
+  });
+
   it('ROLE: LP Deploy', async () => {
     await expect(state.cc.registerLiquidityProvider(state.fund.address)).to.be.reverted;
-    await expect(state.cc.unregister(state.fund.address)).to.be.reverted;
 
     await state.controller.grantRoles(deployer.address, AccessFlags.LP_DEPLOY);
+    await state.cc.unregister(zeroAddress());
+
     await state.cc.registerLiquidityProvider(state.fund.address);
     await state.cc.unregister(state.fund.address);
-    // await state.cc.connect(user2).unregister(user2.address); // cannot test without fund implementing unregister
+    await state.cc.unregister(state.fund.address);
```

Comment:
I was trying to communicate that the ACL role was tested, but CollateralFund does not call unregister to CC so we cannot test the self-unregister path

Reply:
yes, it needs a mock to be registered to test that